collect_app/src/main/java/org/odk/collect/android/formmanagement/ServerFormDownloader.java

```diff
@@ -339,13 +354,13 @@ FileResult downloadXform(String formName, String url, FormDownloaderListener sta
          * SurveyCTO: The file is saved into a temp folder and is moved to the final place if everything
          * is okay, so that garbage is not left over on cancel.
          */
-        private void writeFile(File file, FormDownloaderListener stateListener, InputStream inputStream)
+        private void writeFile(File file, FormDownloaderListener stateListener, InputStream inputStream, File tempDir)
                 throws IOException, InterruptedException {
 
             File tempFile = File.createTempFile(
                     file.getName(),
                     TEMP_DOWNLOAD_EXTENSION,
-                    new File(storagePathProvider.getDirPath(StorageSubdirectory.CACHE))
```

Comment:
@seadowg and I briefly talked about whether the use of this cache directory could have any significance. All files are eventually cleaned up and I don't see any reason why any particular directory has to be used. Passing in the base cache directory, whatever it is, is certainly an improvement.

Reply:
Yeah that's right! I think going one step further and use the Android cache directory (directly) in the dependency declaration (`AppDependencyModule`) would be nice as well.